src/test/codegen/abi-efiapi.rs

```diff
@@ -0,0 +1,20 @@
+// Checks if the correct annotation for the efiapi ABI is passed to llvm.
+
+// compile-flags: -C no-prepopulate-passes
+
+#![crate_type = "lib"]
+#![feature(abi_efiapi)]
+
+// CHECK: define win64 i64 @has_efiapi
+#[no_mangle]
+#[cfg(target_arch = "x86_64")]
```

Comment:
Rather than using `cfg` for tests here, write the test using revision feature. See https://github.com/rust-lang/rust/blob/master/src/test/codegen/optimize-attr-1.rs for an example.